OpenEdXMobile/src/main/java/org/edx/mobile/util/ConfigUtil.kt

```diff
@@ -27,6 +30,56 @@ class ConfigUtil {
             return false
         }
 
+        /**
+         * Utility method to check the Discovery status
+         *
+         * @param environment       Relevant configuration environment.
+         * @return True if Discovery enabled in configurations false else wise
+         */
+        fun isDiscoveryEnabled(environment: IEdxEnvironment? = null): Boolean {
+            return environment?.config?.discoveryConfig?.courseDiscoveryConfig?.isDiscoveryEnabled == true
+        }
```

Comment:
IMO: this would be the better code.

```java
        /**
         * Utility method to check the Course Discovery status
         *
         * @param environment       Relevant configuration environment.
         * @return true if Discovery enabled in configurations false else wise
         */
        fun isDiscoveryEnabled(environment: IEdxEnvironment): Boolean {
            return environment.config.discoveryConfig?.courseDiscoveryConfig?.isDiscoveryEnabled == true
        }
```